private_sharing/api_views.py

```diff
@@ -458,3 +470,102 @@ def post(self, request):
                 data_file.delete()
 
         return Response({"ids": ids}, status=status.HTTP_200_OK)
+
+
+class ProjectCreateAPIView(APIView):
+    """
+    Create a project via API
+
+    Accepts project name, description, and redirect_url as (required) inputs
+
+    The other required fields are auto-populated:
+    is_study:  set to False
+    leader:  set to member.name from oauth2 token
+    coordinator:  get from oauth2 token
+    is_academic_or_nonprofit: False
+    add_data:  false
+    explore_share:  false
+    short_description:  first 139 chars of long_description plus an ellipsis
+    active:  True
+    """
+
+    authentication_classes = (CustomOAuth2Authentication,)
+    permission_classes = (HasValidProjectToken,)
+
+    def get_short_description(self, long_description):
+        """
+        Return first 139 chars of long_description plus an elipse.
+        """
+        if len(long_description) > 140:
+            return "{0}…".format(long_description[0:139])
+        return long_description
+
+    def post(self, request):
+        """
+        Take incoming json and create a project from it
+        """
+        member = get_oauth2_member(request).member
+        serializer = ProjectAPISerializer(data=request.data)
+        if serializer.is_valid():
+            coordinator_join = serializer.validated_data.get("coordinator_join", False)
+            project = serializer.save(
+                is_study=False,
+                is_academic_or_nonprofit=False,
+                add_data=False,
+                explore_share=False,
+                active=True,
+                short_description=self.get_short_description(
+                    serializer.validated_data["long_description"]
+                ),
+                coordinator=member,
+                leader=member.name,
+                request_username_access=False,
+                diyexperiment=True,
```

Comment:
this should default to False and be True if the parameter was provided and is true :)